### ISSUE TITLE
worse and worse performance during finetune

### ISSUE THREAD

**Issue by jiuerbujie** [torrvision/crfasrnn#59]
Hi, thanks for your great work that embeds CRF to DL. I tried your code, actually, the GPU version (https://github.com/hyenal/crfasrnn) and finetuned it on VOC12 train set. I changed the layer "inference1" to "inference1-ft" because I want to relearn the CRF layer. The FCN layers are all fixed.

However, during the training, the mIoU performance on the reduced validation set you give here https://github.com/torrvision/crfasrnn/tree/master/python-scripts gets worse and worse. 

Initialization: 69.259%, even worse than the input of CRF layer (Q0 69.85%)
after 1000 iterations: 69.075%
after 30000 iterations: 67.569%

I also tried to unfix the FCN layers but get the similar results.

I am wondering what is wrong here, can you give me some suggestions? Thank you.

**Comment by bittnt** [comment]
Have you checked your data? CRFasRNN requires input image aligning with ground truth mask. Otherwise, the performance will undoubtedly get worse. From what I understand what you said, it is mostly due to some mistakes in data preparation.

It is also important to tweak the learning rate in the prototxt file, for CRFasRNN layer.

It is also interesting, based on the just fcn-8 bit of the released network, you should expect to get results as follows.

Model	Tes data	mean IOU	mean pix. accuracy	pixel accuracy
FNC-8s-TVG	RV-VOC12	69.85	92.94	78.80

**Comment by jiuerbujie** [comment]
@bittnt , thanks for your fast reply. I have checked my data, the image and ground truth mask is aligned. Further, I use the "IMAGE_SEG_DATA" layer from deeplab to load image and ground truth labels directly from VOC2012's "JPEGImages" and "SegmentationClass" folders to avoid data preparation and make sure nothing is wrong for inputs. But also get the same phenomenon.

According to your suggestion, I will try on the original CPU version and see if it is a bug in the GPU version. Thanks again.

**Comment by KleinYuan** [comment]
@jiuerbujie It seems that I met same issue that results got worse and worse ... not even close to fcn-8s. You got it resolved?
